modules/EmbedPlayer/resources/mw.EmbedPlayerNative.js

```diff
@@ -1452,20 +1453,34 @@
         parseTracks: function(){
             var vid = this.getPlayerElement();
             this.parseAudioTracks(vid, 0); //0 is for a setTimer counter. Try to catch audioTracks, give up after 5 seconds
-            if(this.isLive()) {
-                this.parseTextTracks(vid, 0); //0 is for a setTimer counter. Try to catch textTracks.kind === "metadata", give up after 10 seconds
-            }
+            this.parseTextTracks(vid, 0); //0 is for a setTimer counter. Try to catch textTracks, give up after 10 seconds
         },
         parseTextTracks: function(vid, counter){
             var _this = this;
 	        this.parseTextTracksTimeout = setTimeout(function() {
                 if( vid.textTracks.length > 0 ) {
+	                var textTracksData = {languages: []};
                     for (var i = 0; i < vid.textTracks.length; i++) {
-                        if (vid.textTracks[i].kind === "metadata") {
-                            //add id3 tags support
-                            _this.id3Tag(vid.textTracks[i]);
-                            vid.textTracks[i].mode = "hidden";
-                        }
+                    	var textTrack = vid.textTracks[i];
+	                    if (textTrack.kind === 'metadata') {
+		                    //add id3 tags support
+		                    _this.id3Tag(textTrack);
+	                    } else if (textTrack.kind === 'subtitles' || textTrack.kind === 'caption') {
+		                    textTracksData.languages.push({
+			                    'kind': 'subtitle',
+			                    'language': textTrack.label,
+			                    'srclang': textTrack.label,
+			                    'label': textTrack.label,
+			                    'title': textTrack.label,
+			                    'id': textTrack.id,
+			                    'index': i
+		                    });
+	                    }
+	                    textTrack.mode = 'hidden';
+                    }
+                    if (textTracksData.languages.length) {
```

Comment:
not necessary. i just created it some lines above with languages as empty array (#1462).